### PR TITLE
[DataGrid] Reverse boolean props to default to false

### LINKED ISSUE #155: Default boolean prop value should be false
Baseline rule: https://material-ui.com/guides/api/#property-naming

Issues:
- https://github.com/mui-org/material-ui-x/blob/1e17c0836e2e4831d73db7cec8966c2d82c7d494/packages/grid/x-grid-modules/src/models/gridOptions.tsx#L284
- https://github.com/mui-org/material-ui-x/blob/1e17c0836e2e4831d73db7cec8966c2d82c7d494/packages/grid/x-grid-modules/src/models/gridOptions.tsx#L285
- https://github.com/mui-org/material-ui-x/blob/1e17c0836e2e4831d73db7cec8966c2d82c7d494/packages/grid/x-grid-modules/src/models/gridOptions.tsx#L290

### PR DESCRIPTION
**Pull request by dtassone**
Not sure what the correct behaviour is with checkbox selection...

Close #155

### CODE REVIEW

File: packages/grid/x-grid-modules/src/models/gridOptions.tsx [hunk 1]
```diff
@@ -66,11 +66,11 @@ export interface GridOptions {
    */
   columnBuffer: number;
   /**
-   * Enable multiple selection using the CTRL or CMD key.
+   * Disable multiple selection using the CTRL or CMD key.
```
Comment (mbrookes) [inline packages/grid/x-grid-modules/src/models/gridOptions.tsx:69]:
```suggestion
   * If `true`, multiple selection using the CTRL or CMD key is disabled.
```

Use standard props documentation wording for boolean props. (The "If `true`" part at least - there doesn't seem to be consistency as to whether the outcome should be present or future tense ("is" / "will be")

File: packages/grid/x-grid-modules/src/hooks/features/useSelection.ts [hunk 2]
```diff
@@ -110,9 +110,9 @@ export const useSelection = (
 
   const selectRows = React.useCallback(
     (ids: RowId[], isSelected = true, deSelectOthers = false) => {
-      if (!options.enableMultipleSelection && ids.length > 1) {
+      if (options.disableMultipleSelection && ids.length > 1) {
         throw new Error(
-          'Material-UI: Enable options.enableMultipleSelection to select more than 1 item.',
+          'Material-UI: DisableMultipleSelection should be false to select more than 1 item.',
```
Comment (oliviertassinari) [inline packages/grid/x-grid-modules/src/hooks/features/useSelection.ts:115]:
```suggestion
          'Material-UI: `disableMultipleSelection` should be false to select more than 1 item.',
```

File: packages/grid/x-grid-modules/src/models/gridOptions.tsx (hunk 1 above)
Comment (oliviertassinari) [reply packages/grid/x-grid-modules/src/models/gridOptions.tsx:69]: > present or future tense ("is" / "will be")

I have noticed this inconsistency too yesterday when reviewing the date picker pull request migration to the main repository. Should we settle on the present?

File: packages/storybook/src/documentation/pages/selection.stories.mdx [hunk 3]
```diff
@@ -9,13 +9,13 @@ import DisabledCellSelectionDemo from './demos/selection/disabledCellSelection.d
...
 
-By default, the grid allows multiple selection. However, this setting can be turned off using the property `enableMultipleSelection` of the `options` React prop.
+By default, the grid allows multiple selection. However, this setting can be turned off using the property `disableMultipleSelection` of the `options`React prop.
 
 ```tsx
 <XGrid
   rows={rows}
   columns={columns}
-  options={{ enableMultipleSelection: false }}
+  options={{ disableMultipleSelection: true }}
```
Comment (oliviertassinari) [inline packages/storybook/src/documentation/pages/selection.stories.mdx:18]: Would it conflict with #153? Should we merge it first?

File: packages/grid/x-grid-modules/src/models/gridOptions.tsx [hunk 4]
```diff
@@ -66,43 +66,43 @@ export interface GridOptions {
...
+  disableExtendRowFullWidth?: boolean;
   /**
    * The order of the sorting sequence.
    *
    * @default ['asc', 'desc', null]
    */
   sortingOrder: SortDirection[];
   /**
-   * Activate pagination.
+   * If `true`, pagination is enabled
```
Comment (oliviertassinari) [inline packages/grid/x-grid-modules/src/models/gridOptions.tsx:105]:
```suggestion
   * If `true`, pagination is enabled.
```

File: packages/grid/x-grid-modules/src/models/gridOptions.tsx [hunk 5]
```diff
@@ -147,37 +147,37 @@ export interface GridOptions {
    */
   sortingMode?: FeatureMode;
   /**
-   * Toggle footer component visibility.
+   * If `true`, the footer component is hidden
```
Comment (oliviertassinari) [inline packages/grid/x-grid-modules/src/models/gridOptions.tsx:150]:
```suggestion
   * If `true`, the footer component is hidden.
```

File: packages/grid/x-grid-modules/src/models/gridOptions.tsx [hunk 6]
```diff
@@ -147,37 +147,37 @@ export interface GridOptions {
...
   /**
-   * Toggle footer selected row count element visibility.
+   * If `true`, the selected row count in the footer is hidden
    *
    * @default false
    */
   hideFooterSelectedRowCount?: boolean;
   /**
-   * Toggle footer pagination component visibility.
+   * If `true`, the pagination component in the footer is hidden
```
Comment (oliviertassinari) [inline packages/grid/x-grid-modules/src/models/gridOptions.tsx:168]:
```suggestion
   * If `true`, the pagination component in the footer is hidden.
```

File: packages/grid/x-grid-modules/src/models/gridOptions.tsx [hunk 7]
```diff
@@ -276,18 +276,14 @@ export interface GridOptions {
  * The default [[GridOptions]] object that will be used to merge with the 'options' passed in the react component prop.
  */
 export const DEFAULT_GRID_OPTIONS: GridOptions = {
-  autoHeight: false,
   rowHeight: 52,
   headerHeight: 56,
   scrollbarSize: 15,
   columnBuffer: 2,
-  enableMultipleSelection: true,
-  enableMultipleColumnsSorting: true,
```
Comment (oliviertassinari) [inline packages/grid/x-grid-modules/src/models/gridOptions.tsx:285]: I wonder if we will need to include the default value back for the automatic generation of the documentation, we will see.

File: packages/grid/x-grid-modules/src/models/gridOptions.tsx [hunk 8]
```diff
@@ -147,37 +147,37 @@ export interface GridOptions {
...
   /**
-   * Toggle footer row count element visibility.
+   * If `true`, the row count in the footer is hidden
    *
    * @default false
    */
   hideFooterRowCount?: boolean;
   /**
-   * Toggle footer selected row count element visibility.
+   * If `true`, the selected row count in the footer is hidden
```
Comment (oliviertassinari) [inline packages/grid/x-grid-modules/src/models/gridOptions.tsx:162]:
```suggestion
   * If `true`, the selected row count in the footer is hidden.
```

File: packages/grid/x-grid-modules/src/models/gridOptions.tsx [hunk 9]
```diff
@@ -147,37 +147,37 @@ export interface GridOptions {
...
   /**
-   * Toggle footer component visibility.
+   * If `true`, the footer component is hidden
    *
    * @default false
    */
   hideFooter?: boolean;
   /**
-   * Toggle footer row count element visibility.
+   * If `true`, the row count in the footer is hidden
```
Comment (oliviertassinari) [inline packages/grid/x-grid-modules/src/models/gridOptions.tsx:156]:
```suggestion
   * If `true`, the row count in the footer is hidden.
```

File: packages/grid/x-grid-modules/src/models/gridOptions.tsx (hunk 7 above)
Comment (dtassone) [reply packages/grid/x-grid-modules/src/models/gridOptions.tsx:285]: Probably not 🤔

**Pull request merged**